src/PhpParser/Node/BetterNodeFinder.php

```diff
@@ -551,6 +551,23 @@ private function resolveNeighborNextStmt(StmtsAwareInterface $stmtsAware, Stmt $
         return $stmt;
     }
 
+    private function resolveNeighborPreviousStmt(StmtsAwareInterface $stmtsAware, Stmt $stmt, ?int $key): ?Node
+    {
+        if ($key === null) {
+            $key = 0;
+        }
+
+        if (! isset($stmtsAware->stmts[$key + 1])) {
+            return $stmtsAware->stmts[$key - 1] ?? null;
+        }
+
+        if ($stmtsAware->stmts[$key + 1]->getStartTokenPos() !== $stmt->getStartTokenPos()) {
+            return $stmtsAware->stmts[$key - 1] ?? null;
+        }
+
+        return $stmt;
```

Comment:
@staabm this to cover stmt just removed on lookup previous stmt

Reply:
or not needed, actually, if next is removed, previous key never changed, I will revert :)